evalml/pipelines/components/estimators/classifiers/lightgbm_classifier.py

```diff
@@ -44,17 +44,19 @@ def __init__(self, boosting_type="gbdt", learning_rate=0.1, n_estimators=100, ma
                       "bagging_freq": bagging_freq,
                       "bagging_fraction": bagging_fraction}
         parameters.update(kwargs)
-
+        lg_parameters = copy.copy(parameters)
+        updates = {'subsample': None, 'subsample_freq': None}
```

Comment:
Wait, so do these need to be `None` if `boosting_type != "rf"? As it stands, they will always be `None` going into lightgbm.

If not:
```python
if boosting_type == "rf":
    lg_parameters.update({'subsample': None, 'subsample_freq': None, 'bagging_freq': 1})
```
And pass that to lightgbm.

Either way, it would be nice to have this impl use 2 dicts instead of 3, harder to make buggy in the future :)

Reply:
If we pass `bagging_freq` or `bagging_fraction` into LightGBM, we need to set `subsample` to None to avoid the warning messages

Reply:
Ah got it. I see their docs say subsample and subsample_freq are aliases for  bagging_fraction and bagging_freq.

In that case what about
```python
if boosting_type == "rf":
    lg_parameters.update({'bagging_freq': 1})
# avoid lightgbm warnings having to do with parameter aliases
if bagging_freq is not None or bagging_fraction is not None:
    lg_parameters.update({'subsample': None, 'subsample_freq': None})
```
or something along those lines?